### PR TITLE
AS-122: parallelize google calls or bundle them into insertBucket

### PR DESCRIPTION
**Pull request by ansingh7115**
_before:_
Screen Shot 2019-11-01 at 4 51 46 PM (1)

_after:_
Screen Shot 2019-11-14 at 11 15 01 AM

In the `after`, `setBucketLogging` and `setBucketLabels` are rolled into `insertBucket`. `insertInitialStorageLog`  and `updateBucketIam` are parallelized

- [ ] **Submitter**: Include the JIRA issue number in the PR description
- [ ] **Submitter**: Check that the **Product Owner** has signed off on any user-facing changes
- [ ] **Submitter**: Make sure Swagger is updated if API changes
  - [ ] **...and Orchestration's Swagger too!**
- [ ] **Submitter**: If updating admin endpoints, also update [firecloud-admin-cli](https://github.com/broadinstitute/firecloud-admin-cli)
- [ ] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [ ] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [ ] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [ ] **Submitter**: Database checks:
  * If PR includes new or changed db queries, include the explain plans in the description
  * Make sure liquibase is updated if appropriate
  * If doing a migration, take a backup of the
  [dev](https://console.cloud.google.com/sql/instances/terraform-qfarbdq3lrexxck5htofjs5z6m/backups?project=broad-dsde-dev&organizationId=548622027621)
  and
  [alpha](https://console.cloud.google.com/sql/instances/terraform-r4caezzc35c4tb7pgdhwkmme4y/backups?project=broad-dsde-alpha&organizationId=548622027621)
  DBs in Google Cloud Console
- [ ] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [ ] Tell your tech lead (TL) that the PR exists if they want to look at it
- [ ] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [ ] **LR** sign off
- [ ] **Assign to submitter** to finalize
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [ ] **Submitter**: Inform other teams of any API changes via Slack and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

**Review by davidangb — COMMENTED** [review]
code looks good ... how well does it work? Do we have automated tests for this and/or have you tested manually?

File: core/src/main/scala/org/broadinstitute/dsde/rawls/Boot.scala [hunk 1]
```diff
@@ -16,7 +16,6 @@ import com.google.pubsub.v1.ProjectTopicName
 import com.readytalk.metrics.{StatsDReporter, WorkbenchStatsD}
```
Comment (ansingh7115) [inline core/src/main/scala/org/broadinstitute/dsde/rawls/Boot.scala:16]: This file's changes is due to the removal of Linebacker

File: project/Dependencies.scala [hunk 2]
```diff
@@ -47,7 +47,7 @@ object Dependencies {
   val googleServicemanagement: ModuleID = "com.google.apis"   % "google-api-services-servicemanagement" % ("v1-rev17-" + googleV)
   val googleDeploymentManager: ModuleID = "com.google.apis"   % "google-api-services-deploymentmanager" % ("v2beta-rev20181207-1.28.0")
   val googleGuava: ModuleID =             "com.google.guava"  % "guava" % "19.0"
-  val googleRpc: ModuleID =               "io.grpc" % "grpc-core" % "1.17.1"
```
Comment (ansingh7115) [inline project/Dependencies.scala:50]: updating grpc to match workbench-libs

**Review by davidangb — COMMENTED** [review]
code syntax looks good to me ... it's failing rawls-build and I'd like to see this pass tests and see perf results!

**Comment by ansingh7115** [comment]
jenkins retest

**Comment by ansingh7115** [comment]
this has not yet been tested on perf - I've booked a time for tomorrow

**Review by davidangb — APPROVED** [review]
:+1: lgtm!  Assuming perf tests look good too

**Comment by ansingh7115** [comment]
Just finished running perf tests - there wasn't a really a difference that I could discern - but its possible that the change is too small? Regardless, reducing and parallelizing google calls is still a good idea so I will go ahead and merge.

**Pull request merged**